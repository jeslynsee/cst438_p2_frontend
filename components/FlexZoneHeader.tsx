import React from "react";
import { ThemedText } from "@/components/ThemedText";
import { ThemedView } from "./ThemedView";
import { TextStyle, ViewStyle } from "react-native";

export default function FlexZoneHeader() {
  return (
    <ThemedView style={$container}>
        <ThemedText style={$heading} type="title">FlexZone:</ThemedText>
        <ThemedText style={$subheading} type="subtitle">the fitness app</ThemedText>
    </ThemedView>
  );
}

const $container: ViewStyle = {
    width: "100%",
    marginTop: 16,
};

const $heading: TextStyle = {
    fontWeight: "bold",
    textAlign: "center",
    color: 'green'
};

const $subheading: TextStyle = {
    fontSize: 18,
    textAlign: "center",
    color: 'green'
};