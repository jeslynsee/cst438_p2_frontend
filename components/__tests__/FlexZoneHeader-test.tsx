import * as React from 'react';
import renderer from 'react-test-renderer';

import FlexZoneHeader from '../FlexZoneHeader';

it(`renders correctly`, () => {
  const tree = renderer.create(<FlexZoneHeader />).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`contains the correct text`, () => {
    const tree = renderer.create(<FlexZoneHeader />).toJSON();
    
    expect(tree.children[0].children[0]).toBe('FlexZone:');
    expect(tree.children[1].children[0]).toBe('the fitness app');
});