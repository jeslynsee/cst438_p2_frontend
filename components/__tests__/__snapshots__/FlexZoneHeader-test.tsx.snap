// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "marginTop": 16,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        {
          "fontSize": 32,
          "fontWeight": "bold",
          "lineHeight": 32,
        },
        undefined,
        undefined,
        undefined,
        {
          "color": "green",
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
    FlexZone:
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        {
          "color": "green",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    the fitness app
  </Text>
</View>
`;
